addons/base_import/models/base_import.py

```diff
@@ -214,16 +297,38 @@ def get_fields(self, model, depth=FIELDS_RECURSION_LIMIT):
                     dict(field_value, name='id', string=_("External ID"), type='id'),
                     dict(field_value, name='.id', string=_("Database ID"), type='id'),
                 ]
+                field_value['field_model'] = field['relation']
             elif field['type'] == 'one2many':
-                field_value['fields'] = self.get_fields(field['relation'], depth=depth-1)
+                field_value['fields'] = self.get_fields_tree(field['relation'], depth=depth-1)
                 if self.user_has_groups('base.group_no_one'):
                     field_value['fields'].append({'id': '.id', 'name': '.id', 'string': _("Database ID"), 'required': False, 'fields': [], 'type': 'id'})
+                field_value['field_model'] = field['relation']
+            else:
+                field_value['field_model'] = model
 
             importable_fields.append(field_value)
 
         # TODO: cache on model?
         return importable_fields
 
+    def _filter_fields_by_types(self, model_fields_tree, header_types):
+        """ Remove from model_fields_tree param all the fields and subfields that do not match the types in header_types
+        :param: (list of dicts) model_fields_tree: Contains recursively all the importable fields of the target model.
+                                                   Generated in "get_fields_tree" method.
+        :param: (list) header_types: Contains the extracted fields types of the current header.
+                                     Generated in "_extract_header_types" method.
+        """
+        most_likely_fields_tree = []
+        for field in model_fields_tree:
+            subfields = field.get('fields')
+            if subfields:
+                filtered_field = copy.deepcopy(field)  # Avoid modifying fields.
```

Comment:
You don't need a deep copy, since the recursive case never modifies the original.
```suggestion
                filtered_field = dict(field)  # Avoid modifying fields.
```